pkg/deploy/controller/deploymentconfig/metrics.go

```diff
@@ -0,0 +1,22 @@
+package deploymentconfig
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// DeploymentConfigControllerSubsystem is how this controller is represented in
+// prometheus metrics.
+const DeploymentConfigControllerSubsystem = "deploymentconfig_controller"
+
+var (
+	deploymentConditionCounter = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Subsystem: DeploymentConfigControllerSubsystem,
+			Name:      "set_condition",
+			Help:      "Gauge measuring changes to deployment conditions per condition, reason and the status",
```

Comment:
@kargakis you can querty unavailable deployment via prometheus console (just query the condition counter and specify time).